src/sdl2_compat.c

```diff
@@ -1070,6 +1070,7 @@ SDL2Compat_InitOnStartupInternal(void)
     SDL3_SetHint("SDL_WINDOWS_DPI_AWARENESS", "unaware");
     SDL3_SetHint("SDL_BORDERLESS_WINDOWED_STYLE", "0");
     SDL3_SetHint("SDL_VIDEO_SYNC_WINDOW_OPERATIONS", "1");
+    SDL3_SetHint("SDL_VIDEO_WAYLAND_SCALE_TO_DISPLAY","1");
```

Comment:
```suggestion
    SDL3_SetHint(SDL_HINT_VIDEO_WAYLAND_SCALE_TO_DISPLAY, "1");
```

Reply:
Also it's probably worth adding a comment describing the effect of this so people reading the code later understand why it was added and when they should set the quirk to turn this off.